src/frontend/js/components/DashBoard/ProtectedRoute.tsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect, type PropsWithChildren } from 'react';
+import { location } from 'utils/indirection/window';
+
+interface ProtectedRouteProps {
+  isAllowed: Boolean;
+  redirectPath: string;
+}
+
+const ProtectedRoute = ({
+  isAllowed,
+  redirectPath,
+  children,
+}: PropsWithChildren<ProtectedRouteProps>) => {
+  useEffect(() => {
+    if (!isAllowed) {
+      location.assign(redirectPath);
+    }
+  }, [isAllowed, redirectPath]);
+
+  return <div>{children}</div>;
```

Comment:
No need to wrap `children` within a `<div />`
```suggestion
  return children;
```